*** src/common/image.cpp	2010-10-24 01:58:20.000000000 -0400
--- src/common/image.cpp	2010-10-24 01:58:39.000000000 -0400
***************
*** 435,452 ****
      // either the high-quality or normal method as specified
      if ( quality == wxIMAGE_QUALITY_HIGH )
      {
!         // We need to check whether we are downsampling or upsampling the image
!         if ( width < old_width && height < old_height )
!         {
!             // Downsample the image using the box averaging method for best results
!             image = ResampleBox(width, height);
!         }
!         else
!         {
!             // For upsampling or other random/wierd image dimensions we'll use
!             // a bicubic b-spline scaling method
!             image = ResampleBicubic(width, height);
!         }
      }
      else    // Default scaling method == simple pixel replication
      {
--- 435,443 ----
      // either the high-quality or normal method as specified
      if ( quality == wxIMAGE_QUALITY_HIGH )
      {
!          // For upsampling or other random/wierd image dimensions we'll use
!          // a bicubic b-spline scaling method
!          image = ResampleBicubic(width, height);
      }
      else    // Default scaling method == simple pixel replication
      {
